generator/internal/language/rusttemplate.go

```diff
@@ -72,7 +72,14 @@ type rustMessageAnnotation struct {
 	MessageAttributes []string
 	DocLines          []string
 	HasNestedTypes    bool
-	BasicFields       []*api.Field
+	// All the fields except OneOfs.
+	BasicFields []*api.Field
+	// The subset of `BasicFields` that are neither maps, nor repeated.
+	SingularFields []*api.Field
+	// The subset of `BasicFields` that are repeated (`Vec<T>` in Rust).
+	RepeatedFields []*api.Field
+	// The subset of `BasicFields` that are maps (`HashMap<K, V>` in Rust).
+	MapFields []*api.Field
```

Comment:
Splitting the basic fields like this, makes it easier to write the mustache templates.